symengine/printer.cpp

```diff
@@ -63,14 +63,14 @@ void StrPrinter::bvisit(const Complex &x) {
 
 void StrPrinter::bvisit(const RealDouble &x) {
     std::ostringstream s;
-    s.precision(std::numeric_limits< double >::digits10);
+    s.precision(DBL_DIG);
     s << x.i;
     str_ = s.str();
 }
 
 void StrPrinter::bvisit(const ComplexDouble &x) {
     std::ostringstream s;
-    s.precision(std::numeric_limits< double >::digits10);
+    s.precision(DBL_DIG);
```

Comment:
P.S. I think the Intel compiler was complaining that `digits10` is not part of `std::numeric_limits<double>`. So for this we should just add some ifdefs, or perhaps there is a way to define `digits10` in `std::numeric_limits<double>` just for Intel in some of our header files.